packages/toolpad-app/src/utils/useSearch.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import { TextField } from '@mui/material';
+import Search from '@mui/icons-material/SearchOutlined';
+
+/**
+ * Creates a search field.
+ * @param elements The elements to search through.
+ * @param  {string=} placeholder The value of the placeholder text
+ * @returns The search field and the filtered elements
+ */
+
+function useSearch<T>(elements: T[], placeholder?: string): [JSX.Element, T[]] {
+  const [searchText, setSearchText] = React.useState('');
+
+  const handleSearchInput = React.useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchText(event.target.value);
+  }, []);
+
+  const filteredElements = React.useMemo(() => {
+    return elements.filter((elem: any) =>
```

Comment:
I remember in some other PR it was mentioned that we should not use abbrevations and instead use full name
```suggestion
    return elements.filter((element: any) =>
      element.name.toLowerCase().includes(searchText.toLowerCase()),
```